### ISSUE TITLE
OSC UI elements are not updating correctly

### ISSUE THREAD

**Issue by cyanit** [surge-synthesizer/surge#703]
**Problems**
- The "Type" selection on the UI does not change when you modify its parameter
- The Wavetable UI doesn't load until you select it in the UI

**Expectations**
The GUI follows the parameter modification and updates according to the value of the modified parameter.

**Reproduction**
In order to reproduce the bugs, take a look at them here:
[![Take a look at this bug](https://img.youtube.com/vi/52h1REqogk8/0.jpg)](https://youtu.be/52h1REqogk8 "Take a look at this bug")

**System Infos**
OS: Windows 10
DAW: Bitwig Studio 2.4
Type: VST2, 64bit
Build: NIGHTLY 2019-03-02-96dac41

**Comment by baconpaul** [comment]
Just a comment on this: I thought for a moment (since I was i this code) that it was just that COscMenu is not in the nonmodparam array. But that's not it. the OSCMenu is weird. If you put bitwig into learn mode and change it, bitwig learns no parameters. So this one is a bit tricky.

The reason is, AFAICS, is that CSnapshotMenu (which is base class of COscMenu) when a menu item is selected sets synth->queue_id and synth->queue_xml

Then SurgeSynth later resolves these in ::loadOscAlgos 

but when it does that - when it modifies the SurgeSynth object directly it does it with this code

```
           storage.getPatch().scene[s].osc[i].type.val.i =
                storage.getPatch().scene[s].osc[i].queue_type;
            storage.getPatch().update_controls(false, &storage.getPatch().scene[s].osc[i]);
```

which doesn't broadcast a parameter change.

So start by re-going down the spiral of how the parameters synth and idle interact. Document it this time. Then make it so that you can actually send a param change to a host when you change oscillator type. Then make it so you can reciprocate.

So alas, not simple.

**Comment by baconpaul** [comment]
327fc7cac2e71d73b4b455ed08cd1f7dc431b9cf Fixed this problem as well as #702 I think but I just didn’t close this; so let me bump this back to 1.6.2, confirm that, and close it if so.

**Comment by baconpaul** [comment]
Yeah confirmed this is resolved by the same fix; closing the issue.
